node/store.go

```diff
@@ -380,32 +379,22 @@ func EncodeChunks(chunks [][]byte) ([]byte, error) {
 //
 // decodeChunks((len(chunks[0]), chunks[0], len(chunks[1]), chunks[1], ...)) = chunks
 func decodeChunks(data []byte) ([][]byte, error) {
-	buf := bytes.NewReader(data)
 	chunks := make([][]byte, 0)
-
-	for {
-		var length uint64
-		err := binary.Read(buf, binary.LittleEndian, &length)
-		if errors.Is(err, io.EOF) {
-			break
-		}
-		if err != nil {
-			return nil, err
+	buf := data
+	for len(buf) > 0 {
+		if len(buf) < 8 {
+			return nil, errors.New("invalid data to decode")
 		}
+		chunkSize := binary.LittleEndian.Uint64(buf)
+		buf = buf[8:]
 
-		chunk := make([]byte, length)
-		_, err = buf.Read(chunk)
-		if errors.Is(err, io.EOF) {
-			break
-		}
-		if err != nil {
-			return nil, err
+		if len(buf) < int(chunkSize) {
+			return nil, errors.New("invalid data to decode")
 		}
+		chunk := buf[:chunkSize]
```

Comment:
It should actually also be much more efficient, as this doesn't need to copy chunks (no memory allocation).